### ISSUE TITLE
Tasks running slow

### ISSUE THREAD

**Issue by jasonmccreary** [laravel-shift/workbench-issues#13]
**Describe the bug**
A few of the tasks, particularly the PHP tasks, take several minutes to run.

---

This is likely due to Docker's file I/O. Which is known to be slow, especially on a _Mac_. Projects with either many files or large files are likely to be affected.

A few things you may do to reduce project size:

- Remove any temporarily files: caches, logs, db dumps, etc
- Temporarily remove the `node_modules` folder
- Temporarily remove your development dependencies by removing the `vendor` folder and running `composer install --no-dev`
- For long-lived repositories, manually run `git gc` to optimize your local project `.git` folder.

We will continue to optimize the volume options using by the Workbench. We're also hopeful Docker will continue to improve this issue.

**Comment by Gompje** [comment]
idea: it would be fantastic if workbench would be  able to exclude certain folders like `node_modules`. 

I know I can delete them, but I do _need them to commit_, they are needed for the pre-commit tasks. And while I can bypass those, know that I can optimize setup docker for faster I/O with Parallels like I did in the past (I'm not using docker anymore) -- I was wondering if it wasn't possible for workbench to just to _not copy at that 1 folder_ :)

context: BIG project, with full js frontend. node_modules is about 800mb. 

Just sharing an idea :)

**Comment by jasonmccreary** [comment]
@Gompje, absolutely. Unfortunately it's a Docker thing. We will continue to look into it as there are other mount types. Unfortunately each one has tradeoffs on the various platforms. So what's fast on Windows, may not be fast on Mac.
